pkg/controller/statefulset/statefulset_resize_pvc.go

```diff
@@ -0,0 +1,164 @@
+					err := spc.objectMgr.UpdateClaim(claimClone)
+					spc.recordClaimEvent("resize", set, pod, claimClone, err)
+					if err != nil {
+						return fmt.Errorf("could not update claim %s: %w", claimName, err)
+					}
+				}
+			} else if !matched {
+				spc.recorder.Eventf(set, v1.EventTypeWarning, "FailedUpdatePVC", "failed to update pvc %s: contains diff other than spec resource, wait pvc to be deleted", claimName)
+				return fmt.Errorf("can not patch pvc %s: contains diff other than spec resource, wait pvc to be deleted", claimName)
+			}
+		}
+	}
+	return nil
+}
+
+func (spc *StatefulPodControl) ClaimsCompatible(set *appsv1beta1.StatefulSet, pod *v1.Pod) (bool, error) {
+	ordinal := getOrdinal(pod)
+	templates := set.Spec.VolumeClaimTemplates
+	for i := range templates {
+		claimName := getPersistentVolumeClaimName(set, &templates[i], ordinal)
+		claim, err := spc.objectMgr.GetClaim(set.Namespace, claimName)
+		switch {
+		case apierrors.IsNotFound(err):
+			klog.V(4).InfoS("Expected claim missing, continuing to pick up in next iteration", "claimName", claimName)
+		case err != nil:
+			return false, fmt.Errorf("could not retrieve claim %s for %s when checking PVC spec", claimName, pod.Name)
+		default:
+			if matched, _ := pvc.CompareWithCheckFn(claim, &templates[i], pvc.PVCNeedExpand); !matched {
+				return false, nil
+			}
+		}
+	}
+	return true, nil
+}
+
+func (ssc *defaultStatefulSetControl) checkOwnedPVCStatus(set *appsv1beta1.StatefulSet, pod *v1.Pod, fn pvc.CheckClaimFn) (bool, error) {
+	templates := set.Spec.VolumeClaimTemplates
+	ordinal := getOrdinal(pod)
+	for i := range templates {
+		claimName := getPersistentVolumeClaimName(set, &templates[i], ordinal)
+		claim, err := ssc.podControl.objectMgr.GetClaim(set.Namespace, claimName)
+		switch {
+		case apierrors.IsNotFound(err):
+			klog.V(4).InfoS("Expected claim missing", "claim", claimName)
+			return false, err
+		case err != nil:
+			klog.V(4).ErrorS(err, "Could not retrieve claim", "claim", claimName, "pod", pod.Name)
+			return false, err
+		default:
+			ready := fn(claim, &templates[i])
+			if !ready {
+				return false, err
+			}
+		}
+	}
+	return true, nil
+}
+
+func (ssc *defaultStatefulSetControl) updatePVCStatus(status *appsv1beta1.StatefulSetStatus, set *appsv1beta1.StatefulSet, pods []*v1.Pod) {
+	templates := set.Spec.VolumeClaimTemplates
+	status.VolumeClaimTemplates = make([]appsv1beta1.VolumeClaimTemplateStatus, len(templates))
+	templateNameMap := map[string]*appsv1beta1.VolumeClaimTemplateStatus{}
+	for i := range templates {
+		status.VolumeClaimTemplates[i].VolumeClaimName = templates[i].Name
+		templateNameMap[templates[i].Name] = &status.VolumeClaimTemplates[i]
+	}
+	for _, pod := range pods {
+		if pod == nil {
+			continue
+		}
+		ordinal := getOrdinal(pod)
+		for i := range templates {
+			claimName := getPersistentVolumeClaimName(set, &templates[i], ordinal)
+			claim, err := ssc.podControl.objectMgr.GetClaim(set.Namespace, claimName)
+			switch {
```

Comment:
since the default is the main logic, change the switch case (IsNotFound, err != nil) to if clause, and continue early so as to reduce indentation